sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/Analyzer.scala

```diff
@@ -60,7 +60,7 @@ object SimpleAnalyzer extends Analyzer(
   },
   new SQLConf().copy(SQLConf.CASE_SENSITIVE -> true))
 
-object FakeV2SessionCatalog extends TableCatalog {
+object FakeV2SessionCatalog extends BaseSessionCatalog {
```

Comment:
if we use `CatalogPlugin`, this class can be simplified a lot.

Reply:
Sure. I got rid of everything except for `initialize()` and `name()`.

Reply:
Actually, this still needs to extend `TableCatalog` because this catalog is used in `org.apache.spark.sql.catalyst.analysis.AnalysisSuite.SPARK-28251: Insert into non-existing table error message is user friendly` test, and `asTableCatalog` fails.